### DISCUSSION TITLE
Geospatial API rejections

### DISCUSSION THREAD

**Discussion by mkawahara22** [BuildingComponentLibrary/bcl-manifest#13] [Q&A] [answered]
Hey all-  Not sure if this is the correct forum for this, but I'm having issues with the geospatial API calls to both the current API and the legacy API.  Seems like the examples provided in the documentation aren't even working.  Not sure if I'm doing something wrong, but the legacy calls were working as of yesterday morning (10/4/2021), 

Taking one of the examples from the documentation:
https://bcl.nrel.gov/api/search/location:39.7,-104.5.json

=> yields 'request rejected'

Any help would be appreciated!

Thanks!

**Comment by kflemin** [comment, marked as answer]
Hi @mkawahara22, 

Thank you for letting us know about this! We have fixed the configuration issue and all API examples now seem to be working. Let me know if you run into any more issues.  Thanks!

**Reply by mkawahara22** [reply]
Thanks for the quick reply!  Looks like the calls are working, but for the weather files- there used to be a "url" property.  Does this not exist anymore?  Or is there another method of downloading these files now?

![image](https://user-images.githubusercontent.com/55699187/136062342-c0be188f-934e-43e3-b81b-b5886d1fff3f.png)

this was based off this call:
`https://bcl.nrel.gov/api/search/location:80222.json?fq=component_tags:"site"`

**Reply by kflemin** [reply]
if you go to this page (https://bcl.nrel.gov/documentation) and look in the examples section -> Download Content (very last thing on the page), you will see an example of the download URL. basically, you just need to know the UUID of the content you want to download.  In your example above, the URL for download would be: https://bcl.nrel.gov/api/download/?uids=f2652100-edc6-0131-65e1-48e0eb16a403

We now support downloading different versions of content (you can see examples on how to do that in the same "Download Content" section at the bottom of the page.  By default, the latest version is returned.  But if it's helpful, I can look into listing the download URL for the latest version in the search results above.

**Reply by mkawahara22** [reply]
Thanks!  I think it would be helpful to have the 'url' for the latest version -  We use it in a way where we just stream the text of a specific file into a variable (epw for example) and do not particularly care about some of the other files in the component.  It would save us a few steps from having to sift through all of the files at once.

Appreciate the help on this!

**Reply by kflemin** [reply]
oh, i see. you would want to download a specific file vs. the entire component or measure?  I'll see what I can do :)

**Reply by kflemin** [reply]
@mkawahara22, I added a url to download individual files as well as a "download URL" to download the latest version of the measure or component, so that it is easier to figure out without typing in another API call. Thanks for the feedback
![urls](https://user-images.githubusercontent.com/2205659/136455878-e4578bde-29ce-4411-bc5e-e9df1514bb1a.png)
!

**Reply by mkawahara22** [reply]
Appreciate it!  However, it seems like those geospatial requests are being rejected again.

**Reply by kflemin** [reply]
they are back up now!

**Reply by mkawahara22** [reply]
Hate to bring this back up again- but the `url` property seems to be missing again.  Any chance we can get this back?  Thanks!

![image](https://user-images.githubusercontent.com/55699187/139302961-5b2f9e43-80bc-4d50-ad33-a5dae04dd9cd.png)

**Reply by kflemin** [reply]
not sure what happened, but I have reindexed and they are showing up now! thanks for letting me know!
